extensions-contrib/druid-deltalake-extensions/src/main/java/org/apache/druid/delta/input/DeltaInputSource.java

```diff
@@ -331,6 +333,16 @@ private CloseableIterator<FilteredColumnarBatch> getTransformedDataIterator(
     );
   }
 
+  private Snapshot getLatestSnapshotForTable(final Table table, final Engine engine)
+  {
+    // Setting the LogStore class loader before calling the Delta Kernel snapshot API is required as a workaround with
+    // the 3.2.0 Delta Kernel because the Kernel library cannot instantiate the LogStore class otherwise. Please see
+    // https://github.com/delta-io/delta/issues/3299 for details. This workaround can be removed once the issue is fixed.
+    Thread.currentThread().setContextClassLoader(LogStore.class.getClassLoader());
```

Comment:
We should reset the context class loader in a finally block. 
```
 ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
    try {
      Thread.currentThread().setContextClassLoader(LogStore.class.getClassLoader());
      return table.getLatestSnapshot(engine);
    }
    finally {
      Thread.currentThread().setContextClassLoader(currCtxCl);
    }
```

Reply:
Thanks, done.